pkg/webui/console/components/gateway-data-form/index.js

```diff
@@ -69,19 +81,74 @@ const validationSchema = Yup.object().shape({
   status_public: Yup.boolean().default(false),
   schedule_downlink_late: Yup.boolean().default(false),
   auto_update: Yup.boolean().default(false),
+  schedule_anytime_delay: Yup.string().required(sharedMessages.validateRequired),
 })
 
 @bind
 class GatewayDataForm extends React.Component {
+  static propTypes = {
+    children: PropTypes.node.isRequired,
+    error: PropTypes.error,
+    formRef: PropTypes.shape({}),
+    initialValues: PropTypes.shape({}),
+    /** React reference to be passed to the form */
+    onSubmit: PropTypes.func.isRequired,
+    /** SubmitBar contents */
+    update: PropTypes.bool,
+  }
+
+  static defaultProps = {
+    formRef: undefined,
+    update: false,
+    error: '',
+    initialValues: {},
+  }
+
+  state = {
+    shouldDisplayWarning: false,
+  }
+
   onSubmit(values, helpers) {
     const { onSubmit } = this.props
     const castedValues = validationSchema.cast(values)
 
     onSubmit(castedValues, helpers)
   }
 
+  decodeDelayValue(value) {
+    const duration = value.split(unitRegexp)[0]
+    const unit = value.split(duration)[1]
+
+    this.setState({ shouldDisplayWarning: this.isNotValidDuration(duration, unit) })
+    return {
+      duration: Number(duration),
+      unit,
+    }
+  }
+  // The value of the duration needs to be returned as a single string
+  encodeDelayValue(duration, unit) {
+    this.setState({ shouldDisplayWarning: this.isNotValidDuration(duration, unit) })
+
+    return duration ? `${duration}${unit}` : ''
+  }
+
+  isNotValidDuration(duration, unit) {
+    const durationLowerBound = 130 // set lower bound duration in ms
```

Comment:
Is this value going to stay that way, or should it be fetched somehow from the backend? If it stays like this, I'd suggest to add it to our `pkg/webui/console/constants`.

Reply:
it cannot be fetched from the backend I believe, because it's hidden somewhere deep in GS. I will add it to constants